cpp/src/arrow/chunked_array.h

```diff
@@ -182,6 +182,9 @@ class ARROW_EXPORT ChunkedArray {
   /// \return Status
   Status ValidateFull() const;
 
+  /// \brief Determine if all chunks are located on the CPU
+  bool IsCpu() const;
```

Comment:
For other places we have something like this, we use `is_cpu()` notation. Is that here CamelCase because it is calculated and not a cheap property?

Reply:
Exactly.

But with my PR we might calculate the device set on the constructor then this becomes a cheap check on the pre-calculated device set. So feel free to name it `is_cpu()` even though it's doing a loop.